src/framework/preferences/js/AuxBuilder.js

```diff
@@ -513,14 +513,25 @@ https://github.com/fluid-project/infusion/raw/main/Infusion-LICENSE.txt
      * initialModel, messageLoader, templateLoader, terms, aliases_prefsEditor, and aliases_enhancer. These grades are
      * consumed and integrated by a `fluid.prefs.builder` component.
      *
+     * @param  {String[]} [requestedPrefs] - The preferences requested by the builder
      * @param  {AuxiliarySchema} schemaToExpand - The auxiliary schema to process
      * @param  {Object} topCommonOptions - component options to apply to the various grade components.
      * @param  {Object} elementCommonOptions - component options mixed into the various
      * @param  {PrimarySchema} mappedDefaults - A Primary Schema for the requested preferences.
      * @return {AuxSchema} - The processed auxiliary schema
      */
-    fluid.prefs.expandSchema = function (schemaToExpand, topCommonOptions, elementCommonOptions, mappedDefaults) {
-        var auxSchema = fluid.copy(schemaToExpand);
+    fluid.prefs.expandSchema = function (requestedPrefs, schemaToExpand, topCommonOptions, elementCommonOptions, mappedDefaults) {
+        var schemaIndex = fluid.indexDefaults("auxSchemaIndex", {
+            gradeNames: "fluid.prefs.auxSchema",
+            indexFunc: "fluid.prefs.auxBuilder.defaultSchemaIndexer"
+        });
+        var auxGradeNames = fluid.prefs.auxBuilder.auxGradesForPrefs(schemaIndex, requestedPrefs);
+        var auxGrades = auxGradeNames.map(function (gradeName) {
+            return fluid.defaults(gradeName).auxiliarySchema;
+        });
+        var mergeArgs = [true, {}].concat(auxGrades).concat(schemaToExpand);
+
+        var auxSchema = fluid.extend.apply(null, mergeArgs);
```

Comment:
Using the spread operator you could rewrite this as:
```suggestion
        var auxSchema = fluid.extend.apply(null, [true, {}, ...auxGrades, ...schemaToExpand]);
```